app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -48,7 +49,11 @@ class WireNotificationManager @Inject constructor(
             //      All of this could be handled inside Kalium,
             //      and Reloaded shouldn't need to call `waitUntilLive`.
             //      Kalium could be smarter
-            coreLogic.getSessionScope(userId).syncManager.waitUntilLive()
+            coreLogic.sessionScope(userId) {
```

Comment:
can you please add some comments here: why is it implemented like that :) 

BTW, why not to put this logic into the kalium, so we can call just 1 command in here

Reply:
> can you please add some comments here: why is it implemented like that :)

Good idea. I'll do that

> BTW, why not to put this logic into the kalium, so we can call just 1 command in here

A command like "force sync, get notifications and drop sync" would violate a bit the `ConnectionPolicy`.
Because we have this concept of `ConnectionPolicy` available to Kalium consumers, so they can control a bit the Sync lifecycle, it's important we don't violate and have some side-effects on use cases.

Maybe in the future we can re-think this if this proves itself unstable or too complicated. Let's see how that goes.